tests/X.PagedList.Tests/AsyncPagedListExtensionsTheories.cs

```diff
@@ -34,6 +35,31 @@ public async Task ToListAsync_ForQueryable_WithoutPageNumber_Works(int pageSize,
         Assert.Equal(1, pagedBlogs.PageNumber);
         Assert.Equal(pageSize, pagedBlogs.PageSize);
     }
+    
+    [Theory]
+    [InlineData(1000, 100, 10, null)]
+    [InlineData(1000, 200, 5, 3)]
+    [InlineData(1000, 300, 4, 4)]
+    public async Task ToListAsync_ForQueryable_With_TotalSetCount_Works(int superSetTotalCount, int pageSize,
+        int expectedCount, int? pageNumber = null)
+    {
+        var fixture = new Fixture();
+        
+        pageNumber = pageNumber.HasValue == false ? 0 : pageNumber;
+        var listPageNumber = pageNumber != 0 ? pageNumber - 1 : pageNumber;
+        var xListPageNumber = pageNumber == 0 ? 1 : pageNumber;
+        
+        var superset = fixture.CreateMany<Blog>(superSetTotalCount).AsQueryable();
+        var pageOfSuperSet = superset.Skip(listPageNumber.Value * pageSize).Take(pageSize).ToList();
+        
+        var pagedBlogs = await superset.ToPagedListAsync(xListPageNumber, pageSize, superset.Count());
```

Comment:
You changed method _ToPagedList_ but in test you use _ToPagedListAsync_

Reply:
nice spot!